src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,35 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_properties_if_not_exist_in_path(output_model, input_model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function returns the document after pruning the path which exists
+    in the paths tuple but not in the input_model
+    """
+    output_document = {"properties": output_model.copy()}
+    input_document = {"properties": input_model.copy()}
+    for path in paths:
+        try:
+            if not path_exists(input_document, path):
+                _prop, resolved_path, parent = traverse(output_document, path)
+                key = resolved_path[-1]
+                del parent[key]
+        except LookupError:
+            pass
```

Comment:
Good to pass here, we can verify the input and output in the test_input_equals_output.